Example/Tests/Classes/LPRequestFactoryTest.m

```diff
@@ -0,0 +1,415 @@
+//
+//  LPRequestFactoryTest.m
+//  Leanplum-SDK_Tests
+//
+//  Created by Grace on 10/8/18.
+//  Copyright © 2018 Leanplum. All rights reserved.
+//
+
+#import <XCTest/XCTest.h>
+#import <OCMock/OCMock.h>
+#import <Leanplum/LPRequestFactory.h>
+#import <Leanplum/LeanplumRequest.h>
+#import <Leanplum/LPRequest.h>
+
+NSString *LP_API_METHOD_START = @"start";
+NSString *LP_API_METHOD_GET_VARS = @"getVars";
+NSString *LP_API_METHOD_SET_VARS = @"setVars";
+NSString *LP_API_METHOD_STOP = @"stop";
+NSString *LP_API_METHOD_RESTART = @"restart";
+NSString *LP_API_METHOD_TRACK = @"track";
+NSString *LP_API_METHOD_ADVANCE = @"advance";
+NSString *LP_API_METHOD_PAUSE_SESSION = @"pauseSession";
+NSString *LP_API_METHOD_PAUSE_STATE = @"pauseState";
+NSString *LP_API_METHOD_RESUME_SESSION = @"resumeSession";
+NSString *LP_API_METHOD_RESUME_STATE = @"resumeState";
+NSString *LP_API_METHOD_MULTI = @"multi";
+NSString *LP_API_METHOD_REGISTER_FOR_DEVELOPMENT = @"registerDevice";
+NSString *LP_API_METHOD_SET_USER_ATTRIBUTES = @"setUserAttributes";
+NSString *LP_API_METHOD_SET_DEVICE_ATTRIBUTES = @"setDeviceAttributes";
+NSString *LP_API_METHOD_SET_TRAFFIC_SOURCE_INFO = @"setTrafficSourceInfo";
+NSString *LP_API_METHOD_UPLOAD_FILE = @"uploadFile";
+NSString *LP_API_METHOD_DOWNLOAD_FILE = @"downloadFile";
+NSString *LP_API_METHOD_HEARTBEAT = @"heartbeat";
+NSString *LP_API_METHOD_SAVE_VIEW_CONTROLLER_VERSION = @"saveInterface";
+NSString *LP_API_METHOD_SAVE_VIEW_CONTROLLER_IMAGE = @"saveInterfaceImage";
+NSString *LP_API_METHOD_GET_VIEW_CONTROLLER_VERSIONS_LIST = @"getViewControllerVersionsList";
+NSString *LP_API_METHOD_LOG = @"log";
+NSString *LP_API_METHOD_GET_INBOX_MESSAGES = @"getNewsfeedMessages";
+NSString *LP_API_METHOD_MARK_INBOX_MESSAGE_AS_READ = @"markNewsfeedMessageAsRead";
+NSString *LP_API_METHOD_DELETE_INBOX_MESSAGE = @"deleteNewsfeedMessage";
+
+@interface LPRequestFactory(UnitTest)
+
+- (id<LPRequesting>)createGetForApiMethod:(NSString *)apiMethod params:(NSDictionary *)params;
+- (id<LPRequesting>)createPostForApiMethod:(NSString *)apiMethod params:(NSDictionary *)params;
+- (BOOL)shouldReturnLPRequestClass;
+
+@end
+
+@interface LPRequestFactoryTest : XCTestCase
+
+@end
+
+@implementation LPRequestFactoryTest
+
+- (void)setUp {
+    // Put setup code here. This method is called before the invocation of each test method in the class.
+}
+
+- (void)tearDown {
+    // Put teardown code here. This method is called after the invocation of each test method in the class.
+}
+
+- (void)testCreateGetForApiMethodLPRequest {
+    LPRequestFactory *reqFactory = [[LPRequestFactory alloc]
```

Comment:
Couple things:
1. Avoid using global objects (like `LPFeatureFlagManager`). Just create a new instance of that class and pass it in.
2. Avoid mocking the class you are testing (in this case, RequestFactory). Mock the external class that controls that method (so create and stub a FeatureFlagManager instead)

Just that slight change to ensure that our tests are running on this class. rest looks good.

Reply:
(and the same for `post`)